Adding Tabs to the docs (#4112)
gravitational/teleport: docs/4.3/installation.md

Release](https://gravitational.com/teleport/download/) page for the most
up-to-date information.

=== "Tarball"

    ```bash
    $ curl https://get.gravitational.com/teleport-v{{ teleport.version }}linux-amd64-bin.tar.gz.sha256
    # <checksum> <filename>
    $ curl -O https://get.gravitational.com/teleport-v{{ teleport.version }}-linux-amd64-bin.tar.gz
    $ shasum -a 256 teleport-v{{ teleport.version }}-linux-amd64-bin.tar.gz
    # Verify that the checksums match
    $ tar -xzf teleport-v{{ teleport.version }}-linux-amd64-bin.tar.gz
    $ cd teleport
    $ ./install
    $ which teleport
    /usr/local/bin/teleport
    ```

> === "Debian"

Reviewer: Should probably be `DEB` (as there are various other distros that aren't Debian but use the `.deb` package format)

Author: Updated

    ```bash
    $ curl https://get.gravitational.com/teleport_{{ teleport.version }}_amd64.deb.sha256
    # <checksum> <filename>
    $ curl -O https://get.gravitational.com/teleport_{{ teleport.version }}_amd64.deb
    $ sha256sum teleport_{{ teleport.version }}_amd64.deb
    # Verify that the checksums match
    $ dpkg -i teleport_{{ teleport.version }}_amd64.deb
    $ which teleport
    /usr/local/bin/teleport
    ```

=== "RPM"

    ```bash
    $ curl https://get.gravitational.com/teleport-{{ teleport.version }}-1.x86_64.rpm.sha256
    # <checksum> <filename>
    $ curl -O https://get.gravitational.com/teleport-{{ teleport.version }}-1.x86_64.rpm
    $ sha256sum teleport-{{ teleport.version }}-1.x86_64.rpm
    # Verify that the checksums match
    $ rpm -i teleport-{{ teleport.version }}-1.x86_64.rpm
    $ which teleport
    /usr/local/bin/teleport
    ```

> === "Amazon Linux 2"

Reviewer: I think Amazon Linux is actually RPM-based - do we need to copy the tarball instructions? They'd be better off installing from RPM.

Author: I moved it to RPM based, and I updated the example to use `wget` since `curl` isn't installed as default on AWS Linux 2

    # <checksum> <filename>
    $ curl -O https://get.gravitational.com/teleport-{{ teleport.version }}-1.x86_64.rpm
    $ sha256sum teleport-{{ teleport.version }}-1.x86_64.rpm
    # Verify that the checksums match
    $ rpm -i teleport-{{ teleport.version }}-1.x86_64.rpm
    $ which teleport
    /usr/local/bin/teleport
    ```

=== "Amazon Linux 2"

    ```bash
    $ wget https://get.gravitational.com/teleport-{{ teleport.version }}-1.x86_64.rpm
    $ tar -xzf teleport-v{{ teleport.version }}-1.x86_64.rpm
    $ cd teleport
    $ sudo ./install
    $ which teleport
    /usr/local/bin/teleport
    ```

=== "ARMv7"

    ```bash
    $ curl https://get.gravitational.com/teleport-v{{ teleport.version }}-linux-arm-bin.tar.gz.sha256
    # <checksum> <filename>
    $ curl -O https://get.gravitational.com/teleport-v{{ teleport.version }}-linux-arm-bin.tar.gz
    $ shasum -a 256 teleport-v{{ teleport.version }}-linux-arm-bin.tar.gz
    # Verify that the checksums match
    $ tar -xzf teleport-v{{ teleport.version }}-linux-arm-bin.tar.gz
    $ cd teleport
    $ ./install
    $ which teleport
    /usr/local/bin/teleport
    ```

## Docker

> Please follow our [OSS Docker Quickstart](docs/4.3/quickstart_docker.md) or [Enterprise Docker Quickstart](enterprise/quickstart-enterprise/#run-teleport-enterprise-using-docker) for install and setup instructions.

Reviewer: ```suggestion
Please follow our [OSS Docker Quickstart](quickstart_docker.md) or [Enterprise Docker Quickstart](enterprise/quickstart-enterprise/#run-teleport-enterprise-using-docker) for install and setup instructions.
```

(fix docs link, whoops, my bad)

Author: Can you make this `enterprise/quickstart-enterprise.md#run-teleport-enterprise-using-docker`

Reviewer: Done

=== "Amazon Linux 2"

    ```bash

>     $ wget https://get.gravitational.com/teleport-{{ teleport.version }}-1.x86_64.rpm
>     $ tar -xzf teleport-v{{ teleport.version }}-1.x86_64.rpm

Reviewer: ```suggestion
    $ yum install https://get.gravitational.com/teleport-{{ teleport.version }}-1.x86_64.rpm
```

Author: `curl` and `sha256sum` aren't available on AWS boxes by default.

Reviewer: OK, as there's no default way to check the checksum anyway then how about we just install direct with `yum`? I'll change the suggestion appropriately.

Reviewer: We could also `wget` and then `rpm -i` if that's more consistent with the other RPM method we list.

Author: Yes, please change the suggestion to proposed `yum` based flow.

    ```

## Docker

> Please follow our [Docker Quickstart](https://github.com/gravitational/teleport/tree/master/examples/chart/teleport) for install and setup instructions.

Reviewer: ```suggestion
Please follow our [OSS Docker Quickstart](docs/4.3/quickstart_docker.md) or [Enterprise Docker Quickstart](enterprise/quickstart-enterprise/#run-teleport-enterprise-using-docker) for install and setup instructions.
```

(fix links - it was pointing to the Helm chart before)

## MacOS

=== "Homebrew"

    ```bash
    $ brew install teleport
    ```

=== "Download"

      [Download MacOS .pkg installer](https://get.gravitational.com/tsh-{{ teleport.version }}.pkg) (tsh client only, signed) file, double-click to run the Installer.

> 

Reviewer: ```suggestion

!!! note
    This method only installs the `tsh` client for interacting with Teleport clusters. If you need the `teleport` server or `tctl` admin tool, use the "Terminal" method instead.
```

=== "Terminal"

    ```bash
    $ curl -O https://get.gravitational.com/teleport-{{ teleport.version }}.pkg
    $ sudo installer -pkg teleport-{{ teleport.version }}.pkg -target / # Installs on Macintosh HD
    Password:
    installer: Package name is teleport-{{ teleport.version }}
    installer: Upgrading at base path /
    installer: The upgrade was successful.
    $ which teleport
    /usr/local/bin/teleport
    ```

## Windows (tsh client only)

As of version v3.0.1 we have `tsh` client binary available for Windows 64-bit
architecture - `teleport` and `tctl` are not supported.

> === "Poweshell"

Reviewer: ```suggestion
=== "Powershell"
```

    ```bash
    > curl https://get.gravitational.com/teleport-v{{ teleport.version }}-windows-amd64-bin.zip.sha256
    # <checksum> <filename>

>     > curl -O https://get.gravitational.com/teleport-v{{ teleport.version }}-windows-amd64-bin.zip

Reviewer: ```suggestion
    > curl -O teleport-v{{ teleport.version }}-windows-amd64-bin.zip https://get.gravitational.com/teleport-v{{ teleport.version }}-windows-amd64-bin.zip
```

(you have to explicitly provide an output path otherwise it doesn't save the file)
